br/pkg/lightning/common/common.go

```diff
@@ -89,17 +89,26 @@ func getGlobalAutoIDAlloc(store kv.Storage, dbID int64, tblInfo *model.TableInfo
 	hasAutoIncID := tblInfo.GetAutoIncrementColInfo() != nil
 	hasAutoRandID := tblInfo.ContainsAutoRandomBits()
 
-	// Current TiDB has some limitations for auto ID.
+	// Before MySQL compatible AUTO_INCREMENT, TiDB has some limitations:
 	// 1. Auto increment ID and auto row ID are using the same RowID allocator.
 	//    See https://github.com/pingcap/tidb/issues/982.
 	// 2. Auto random column must be a clustered primary key. That is to say,
 	//    there is no implicit row ID for tables with auto random column.
 	// 3. There is at most one auto column in a table.
 	// Therefore, we assume there is only one auto column in a table and use RowID allocator if possible.
+	//
+	// If AUTO_ID_CACHE=1 is set (i.e. MySQL compatible AUTO_INCREMENT), the limitations do not hold anymore.
 	switch {
-	case hasRowID || hasAutoIncID:
+	case hasRowID:
```

Comment:
i have a unfinished pr to fix this https://github.com/pingcap/tidb/issues/46100 on lightning part

this won't work as lightning don't rebase this alloctor when running in single instance

Reply:
Get it, I'll reset the change here.